CMakeLists.txt

```diff
@@ -111,13 +111,24 @@ endif()
 
 include(${HPCC_SOURCE_DIR}/cmake_modules/commonSetup.cmake)
 
-if ( MAKE_R_ONLY )
+if ( PLUGIN )
     HPCC_ADD_SUBDIRECTORY (roxie/roxiemem)
     HPCC_ADD_SUBDIRECTORY (rtl/nbcd)
     HPCC_ADD_SUBDIRECTORY (rtl/eclrtl)
     HPCC_ADD_SUBDIRECTORY (system/lzma)
     HPCC_ADD_SUBDIRECTORY (system/jlib)
-    HPCC_ADD_SUBDIRECTORY (plugins/Rembed)
+    HPCC_ADD_SUBDIRECTORY (dali/base)
+    HPCC_ADD_SUBDIRECTORY (common/workunit "CASSANDRAEMBED")
+    HPCC_ADD_SUBDIRECTORY (plugins/Rembed "REMBED")
+    HPCC_ADD_SUBDIRECTORY (plugins/v8embed "V8EMBED")
+    HPCC_ADD_SUBDIRECTORY (plugins/memcached "MEMCACHED")
+    HPCC_ADD_SUBDIRECTORY (plugins/pyembed "PYEMBED")
+    HPCC_ADD_SUBDIRECTORY (plugins/redis "REDISEMBED")
```

Comment:
Redis (like memcached and Kafka) is not really "embedded". For consistency either MEMCACHED flag should be MEMCACHEDEMBED, or else (I think preferably) memcached, redis, and kafka should all use the suffix "PLUGIN" rather than EMBED.